votor/src/consensus_pool/stats.rs

```diff
@@ -24,11 +19,45 @@ pub(crate) struct ConsensusPoolStats {
 
     pub(crate) new_certs_generated: Vec<u32>,
     pub(crate) new_certs_ingested: Vec<u32>,
-    pub(crate) ingested_votes: Vec<u32>,
+    ingested_votes: VoteStats,
 
     pub(crate) last_request_time: Instant,
 }
 
+#[derive(Default)]
+struct VoteStats {
+    notarize: u64,
+    finalize: u64,
+    skip: u64,
+    notarize_fallback: u64,
+    skip_fallback: u64,
+}
+
+impl VoteStats {
+    fn increment(&mut self, vote: &Vote) {
+        match vote {
+            Vote::Notarize(_) => self.notarize = self.notarize.saturating_add(1),
+            Vote::NotarizeFallback(_) => {
+                self.notarize_fallback = self.notarize_fallback.saturating_add(1)
+            }
+            Vote::Skip(_) => self.skip = self.skip.saturating_add(1),
+            Vote::SkipFallback(_) => self.skip_fallback = self.skip_fallback.saturating_add(1),
+            Vote::Finalize(_) => self.finalize = self.finalize.saturating_add(1),
+        }
+    }
+
+    fn record(&self, header: &'static str) {
+        datapoint_info!(
+            header,
+            ("finalize", self.finalize, i64),
+            ("notarize", self.notarize, i64),
+            ("notarize_fallback", self.notarize_fallback, i64),
+            ("skip", self.skip, i64),
+            ("skip_fallback", self.skip_fallback, i64),
```

Comment:
nit: I guess this is retaining the existing names, but `num_finalize_votes`, `num_notarize_votes`, etc might read better. That being said, the extra info that `num_*_votes` adds can likely be deduced. I don't feel too strongly about this, I guess we can see what others think

Reply:
I like the suggestion.